sherpa/models/model.py

```diff
@@ -1430,20 +1445,12 @@ def cache_status(self) -> None:
     __pow__, __rpow__ = _make_binop(np.power, '**')
 
     def __setstate__(self, state):
-        self.__dict__.update(state)
-
+        # In case old pickle files do not have cache info
```

Comment:
Given the discussion in `cache_clear` above, I would think we should only call `cache_clear` after checking the `_use_caching` field. Technically it should not matter as written, but I think logically it would make sense.

Reply:
In the final version of this PR, there is no `_use_caching` any more. This is more of a compatibility layer for old pickle files. In fact, it might be better to never pickle (or unpickle) the cache at all, because the risk that some hidden state (such as the ATOMDB version or the elemental abundances) have inadvertently changed, not because the user changed a model, but because e.g. the Sherpa version was upgraded. That's bad enough on it's own, but then to have cache values that are restored invites crazy inconsistencies that throw a fit off.

Reply:
I can helpfully come up with arguments for both sides, so I'm not sure I'm much help here ...

Reply:
OK then, I have arguments for clearing the cache, so that's go with that.